tests/integration/test_vulnerability_detector/test_providers/test_providers_no_os.py

```diff
@@ -57,115 +57,161 @@
     - providers
 '''
 import os
-
 import pytest
-from wazuh_testing.modules import vulnerability_detector as vd
-from wazuh_testing.tools import LOG_FILE_PATH
-from wazuh_testing.tools.configuration import load_wazuh_configurations, check_apply_test
-from wazuh_testing.tools.monitoring import FileMonitor
-from wazuh_testing.tools.services import control_service
-# from wazuh_testing.vulnerability_detector import make_vuln_callback, VULN_DETECTOR_EXTENDED_GLOBAL_TIMEOUT
-# Disabled until refactor
 
+from datetime import date
+from wazuh_testing.db_interface import cve_db
+from wazuh_testing.tools.services import control_service
+from wazuh_testing.tools import configuration
+from wazuh_testing.modules import vulnerability_detector as vd
+from wazuh_testing.modules.vulnerability_detector import event_monitor as evm
 
 # Marks
-pytestmark = [pytest.mark.server, pytest.mark.tier(level=1)]
-
-# variables
-test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "data")
-configurations_path = os.path.join(test_data_path, "wazuh_providers_os.yaml")
+pytestmark = [pytest.mark.server]
+
+TEST_DATA_PATH = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+CONFIGURATIONS_PATH = os.path.join(TEST_DATA_PATH, 'configuration_template')
+TEST_CASES_PATH = os.path.join(TEST_DATA_PATH, 'test_cases')
+
+configurations_path = os.path.join(CONFIGURATIONS_PATH, 'providers_no_os.yaml')
+test_enabled_cases_path = os.path.join(TEST_CASES_PATH, 'test_no_os.yaml')
+configuration_parameters, configuration_metadata, test_case_ids = configuration.get_test_cases_data(
+    test_enabled_cases_path)
+configurations = configuration.load_configuration_template(configurations_path, configuration_parameters,
+                                                           configuration_metadata)
+
+# Add <update_from_year> tag for NVD to download only the last year feed and speed up the testing process
+for index in range(len(configurations)):
+    if configurations[index]['sections'][0]['elements'][1]['provider']['attributes'][0]['name'] == 'nvd':
+        configurations[index]['sections'][0]['elements'][1]['provider']['elements'][0]['update_from_year'] = \
```

Comment:
`=` or `append`? Only the NVD provider should have that tag (`update_from_year`)

Reply:
- NVD provider is the only one that uses the tag `update_from_year` and it is specified in this [link](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/vuln-detector.html#provider)
- Using the `=` in the for loop It get the follow configuration:

`<vulnerability-detector>
    <enabled>yes</enabled>
    <provider name="nvd">
      <enabled>yes</enabled>
      <update_from_year>2022</update_from_year>
    </provider>
  </vulnerability-detector>
`

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/2547/commits/cf0038f5ef58427ecede8adbd597b20900631b83